File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v3/entity/Post.java
package com.example.dtomappingbenchmark.v3.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity(name = "PostV3")
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Long id;

    @Column
    private String title;
    @Column
    private String content;

    @ManyToOne(fetch = FetchType.LAZY) // 지연 로딩을 이용하기 위해
    private User user;
}

File: object-mapping-benchmark-v5/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv5/post/entity/Post.kt
package yu.selab.cj.objectmappingbenchmarkv5.post.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import yu.selab.cj.objectmappingbenchmarkv5.user.entity.User


@Entity
class Post(

    @Column
    val title: String,

    @Column
    val content: String?,

    @ManyToOne
    @JoinColumn(name = "uid")
    var user: User
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val pid : Long = 0

}